ports/nrf/supervisor/port.c

```diff
@@ -264,7 +267,17 @@ void port_sleep_until_interrupt(void) {
         sd_app_evt_wait();
     } else {
         // Call wait for interrupt ourselves if the SD isn't enabled.
-        __WFI();
+        // Note that `wfi` should be called with interrupts disabled,
+        // to ensure that the queue is properly drained.  The `wfi`
+        // instruction will returned as long as an interrupt is
+        // available, even though the actual handler won't fire until
+        // we re-enable interrupts.
+        common_hal_mcu_disable_interrupts();
+        if (!tud_task_event_ready()) {
```

Comment:
Thanks @xobs it look super clean now :+1: :+1:  , @tannewt we should include the check for other background task that uses event queue mechanism like TinyUSB as well in the future. This should be applied to all other ports as well.